### ISSUE TITLE
BWA ALN driving to a 'EOF error' on some FASTQ

### ISSUE THREAD

**Issue by PWSmit** [SouthGreenPlatform/TOGGLE#8]
Hi, 

I tried to use TOGGLE with the test data provided and with my own files. the EOF error pops-up with every version of BWA I tried. I have tried 7.12, 7.15, 6.2, and 7.13. All of them give me the same error. 
Suggestion which version I should use would be appreciated. Thanks! 

p.s. the automatic installer, doesn't work. it results in various permission errors and end-up with a practically empty folder...

**Comment by francoissabot** [comment]
Hi
Did you try the incriminate file directly using BWA aln w/o TOGGLE ? There is a weird bug in aln (not at all TOGGLE related) that sometimes provokes such issue, but we do not know why.

Either, try using bwa mem instead of aln/sampe.

For the installer, can you provide me the errors and the config (system, kernel, gcc...) that I can check ?

cheers

F

**Comment by PWSmit** [comment]
Hi,

Thanks for the suggestion.

 I used BWA mem directly w/o TOGGLE and it works OK. However, via TOGGLE I end up with the same error using BWA mem. in the terminal, he gives me the same error again as well: 
“use of uninitialized value $waitOutput in numeric ne (!=) at home/manager/TOGGLE/toggleGenerator.pl line 541”. It does produce a SAM file which looks OK to me. 

I couldn’t find the reason for the error, but I didn’t wrote the tool :) 
Your help will be greatly appreciated as I have to run 250 isolates and your tool would be great for it! 

Many thanks, 

Pieter

**Comment by PWSmit** [comment]
Hi Francois, 

When I redid the TOGGLE analysis, I took a closer (read: better) look at what BWA mem generates, which is a sam file rather than a BAM file. When reading up on this online I found this website which explained the issue: 
https://www.biostars.org/p/186882/ https://www.biostars.org/p/186882/ (at the bottom is the answer). adding -S to the script for samtools view  (so samtools view -b -h -S) does the trick! The pipeline completes OK and I get a vcf file produced with some variants (n=15) being called using the SNPdiscorveryPairedcontig.txt, with the TwopairedIndividusIrigin datasets. 

so it works again, many thanks for your input! 

Pieter
